### ISSUE TITLE
false positive in OrphanedFormatString check for non-traditional URL schemes

### ISSUE THREAD

**Issue by erm-g** [google/error-prone#4609]
**What version of Error Prone are you using?**
Latest - 2.33.0

**Does this issue reproduce with the latest release?**
Yes

**What did you do?**

``` java
public static void main(String[] args) {
    StringBuilder url = new StringBuilder("https://mydomain%21com/");
    url.append("path");
    System.out.println(url);
    url = new StringBuilder("spiffe://mydomain%21com/");
    url.append("path");
    System.out.println(url);
}
```
**What did you expect to see?**
No warning

**What did you see instead?**
`[9,29] [OrphanedFormatString] String literal contains format specifiers, but is not passed to a format method`
[9,29] indicates "spiffe://" case, not the  "https://" one
